src/test/java/spoon/test/template/SubstitutionTest.java

```diff
@@ -47,4 +51,44 @@ public void testSubstitutionInsertAllFields() {
         @Override
         public void statement() { }
     }
+
+    @Test
+    public void testSubstitutionInsertAllNestedTypes() {
+        // contract: Substitution.insertAllNestedTypes inserts the only nested class from a singly nested template into the target class
+
+        // arrange
+        Launcher spoon = new Launcher();
+        spoon.addTemplateResource(new FileSystemFile("./src/test/java/spoon/test/template/SubstitutionTest.java"));
+
+        spoon.buildModel();
+        Factory factory = spoon.getFactory();
+
+        CtType<?> targetType = factory.Class().create("goodClass");
+        StatementTemplate template = new SinglyNestedTemplate();
+
+        // act
+        Substitution.insertAllNestedTypes(targetType, template);
+
+        // assert
+        assertEquals(1, targetType.getNestedTypes().size());
```

Comment:
This assertion is a bit too weak. At least check that the inserted type has the expected name.

Reply:
Yes sure, I will push a commit to rectify that